components/gitpod-cli/cmd/rebuild.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor"
+	"github.com/gitpod-io/gitpod/gitpod-cli/pkg/utils"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/spf13/cobra"
+)
+
+func TerminateExistingContainer() error {
+	cmd := exec.Command("docker", "ps", "-q", "-f", "label=gp-rebuild")
+	containerIds, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	for _, id := range strings.Split(string(containerIds), "\n") {
+		if len(id) == 0 {
+			continue
+		}
+
+		cmd = exec.Command("docker", "stop", id)
+		err := cmd.Run()
+		if err != nil {
+			return err
+		}
+
+		cmd = exec.Command("docker", "rm", "-f", id)
+		err = cmd.Run()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var buildCmd = &cobra.Command{
+	Use:    "rebuild",
+	Short:  "Re-builds the workspace image (useful to debug a workspace custom image)",
+	Hidden: false,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		client, err := supervisor.New(ctx)
+		if err != nil {
+			utils.LogError(ctx, err, "Could not get workspace info required to build", client)
+			return
+		}
+		defer client.Close()
+
+		tmpDir, err := os.MkdirTemp("", "gp-rebuild-*")
+		if err != nil {
+			log.Fatal("Could not create temporary directory")
+			return
+		}
+		defer os.RemoveAll(tmpDir)
+
+		wsInfo, err := client.Info.WorkspaceInfo(ctx, &api.WorkspaceInfoRequest{})
+		if err != nil {
+			utils.LogError(ctx, err, "Could not fetch the workspace info", client)
+			return
+		}
+
+		event := utils.TrackEvent(ctx, client, &utils.TrackCommandUsageParams{
```

Comment:
Should not we send it in defer after recover to track all usages?

Reply:
Alternatively make top function very simple and call to another function which returns an event and an error as a result. It would have handle fatal logging as well

Reply:
Good call, moved to `defer event.Send(ctx)` at the top, so it covers all usages.